api/v1alpha1/api.go

```diff
@@ -1,3 +1,122 @@
+	//
+	// Currently, the only supported schema is OpenAI as the input schema.
+	APISchema LLMAPISchema `json:"inputSchema"`
+	// TargetRefs are the names of the Gateway resources this policy is being attached to.
+	// The namespace is "local", i.e. the same namespace as the LLMRoute.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=128
+	TargetRefs []gwapiv1a2.LocalPolicyTargetReferenceWithSectionName `json:"targetRefs"`
+	// BackendRefs lists the LLMBackends that this LLMRoute will route traffic to.
+	// The namespace is "local", i.e. the same namespace as the LLMRoute.
+	//
+	// +kubebuilder:validation:MaxItems=128
+	BackendRefs []LLMBackendLocalRef `json:"backendRefs,omitempty"`
+}
+
+// LLMBackendLocalRef is a reference to a LLMBackend resource in the "local" namespace.
+type LLMBackendLocalRef struct {
+	// Name is the name of the LLMBackend in the same namespace as the LLMRoute.
+	Name string `json:"name"`
+}
+
 // +kubebuilder:object:root=true
+
+// LLMBackend is a resource that represents a single backend for LLMRoute.
+// A backend is a service that handles traffic with a concrete API specification.
+type LLMBackend struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines the details of the LLM policy.
+	Spec LLMBackendSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// LLMBackendList contains a list of LLMBackends.
+type LLMBackendList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []LLMBackend `json:"items"`
+}
+
+// LLMBackendSpec details the LLMBackend configuration.
+type LLMBackendSpec struct {
+	// APISchema specifies the API schema of the output format of requests from
+	// Envoy that this LLMBackend can accept as incoming requests.
+	// Based on this schema, the ai-gateway will perform the necessary transformation for
+	// the pair of LLMRouteSpec.APISchema and LLMBackendSpec.APISchema.
+	//
+	// This is required to be set.
+	APISchema LLMAPISchema `json:"outputSchema"`
+}
+
+// LLMAPISchema defines the API schema of either LLMRoute (the input) or LLMBackend (the output).
+//
+// This allows the ai-gateway to understand the input and perform the necessary transformation
+// depending on the API schema pair (input, output).
+//
+// Note that this is vendor specific, and the stability of the API schema is not guaranteed by
+// the ai-gateway, but by the vendor via proper versioning.
+type LLMAPISchema struct {
+	// Schema is the API schema of the LLMRoute or LLMBackend.
+	//
+	// +kubebuilder:validation:Enum=OpenAI;AWSBedrock
+	Schema APISchema `json:"schema"`
+
+	// Version is the version of the API schema.
+	Version string `json:"version,omitempty"`
+}
+
+// APISchema defines the API schema.
+type APISchema string
+
+const (
+	// APISchemaOpenAI is the OpenAI schema.
+	//
+	// https://platform.openai.com/docs/overview
```

Comment:
In the case of OpenAI, they publish an OpenA*P*I spec, so maybe that's the more helpful link?

```suggestion
	// https://github.com/openai/openai-openapi
```